src/main/java/ch/fhnw/algd1/converters/binary/BinConverter.java

```diff
@@ -2,14 +2,31 @@
 
 public class BinConverter {
 	public static String toString(int x) {
-		// TODO expect x to be in range [-128, 127], return string with 8 binary
-		// digits representing x in 2-complement
-		return "00000000";
+		String s = "";
+		if (x < 0) {
+			x += 128;
+			s += 1;
+		} else {
+			s += 0;
+		}
+
+		for (int i = 6; i >= 0; i--) {
+			s += (x >> i) & 1; // add the (i)th from right bit
+		}
+
+		return s;
 	}
 
 	public static int fromString(String text) {
-		// TODO expect text to contain 8 binary digits, parse to int value in
-		// 2-complement
-		return 0;
+		int result = 0;
+		for (int i = 1; i < 8; i++) {
+			if (text.charAt(i) == '1') {
+				result += (1 << (7 - i));
```

Comment:
Ja gut, kann man so machen, ist jetzt aber etwas "bastelig" in dem Sinne, dass hier eine `7` steht (also eigentlich eine `8 - 1`, was relevant wäre, wenn man jetzt die `8` (also die String-Länge) als Parameter des Verfahrens haben wollen würde). Mit dem Ansatz, den ich heute noch kurz an der Wandtafel gezeigt habe, liesse sich die Aufgabe sogar lösen, ohne dass hier die Anzahl der Ziffern überhaupt bekannt sein muss... Nehmen Sie diese Challenge an :-)?